NuKeeper.Inspection/NuGetApi/ConcurrentSourceRepositoryCache.cs

```diff
@@ -0,0 +1,23 @@
+using System.Collections.Concurrent;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuKeeper.Inspection.NuGetApi
+{
+    public class ConcurrentSourceRepositoryCache
+    {
+        private readonly ConcurrentDictionary<PackageSource, SourceRepository> _packageSources
+            = new ConcurrentDictionary<PackageSource, SourceRepository>();
+
+        public SourceRepository Get(PackageSource source)
+        {
+            return _packageSources.GetOrAdd(source, CreateSourceRepository);
+        }
+
+        private static SourceRepository CreateSourceRepository(PackageSource packageSource)
+        {
+            return new SourceRepository(packageSource, Repository.Provider.GetCoreV3());
```

Comment:
This line is equivalent to before, but could be changed in a later PR.
The issue is that it always adds the source as a Nuget protocol V3 source.
But you can use the property `int PackageSource.ProtocolVersion` to check.
I don't know what you would do if it's V2 or something else though.